src/utility/hexgrid.js

```diff
@@ -430,29 +432,66 @@ export class HexGrid {
 				flipped: false,
 				id: 0,
 				team: Team.enemy,
+				replaceEmptyHexesWithDashed: false,
 			};
 
 		o = $j.extend(defaultOpt, o);
 
-		// Exclude everything but the creatures
-		o.hexes = o.hexes.filter((hex) => {
-			if (hex.creature instanceof Creature && hex.creature.id != o.id) {
-				if (!o.optTest(hex.creature)) {
-					return false;
+		/* Divide hexes into:
+		- containing valid targets
+		- empty (no possible target)
+		Hexes containing invalid targets (wrong team, o.optTest, etc) are discard. */
+		const { targetHexes, emptyHexes } = o.hexes.reduce(
```

Comment:
This is the code I mentioned leaving in, although it's not currently use. It separates the potentially target hexes into valid targets, ignored (invalid targets), and empty. In the future the empty hexes could be filled with the solid black hexagons.

Reply:
Yeah, did some testing, I see what you meant :)